dandelion-voting/contracts/DandelionVoting.sol

```diff
@@ -225,6 +226,31 @@ contract DandelionVoting is IForwarder, AragonApp {
         return canPerform(_sender, CREATE_VOTES_ROLE, arr());
     }
 
+    // ACL Oracle fns
+
+    /**
+    * @notice Returns whether the sender has voted on the most recent open vote or closed unexecuted vote.
+    * @dev IACLOracle interface conformance. The ACLOracle permissioned function should specify the sender
+    *      with 'authP(SOME_ACL_ROLE, arr(sender))', where sender is typically set to 'msg.sender'.
+    * @param _who The address to check if can perform (ignored if `_how` contains an address)
+    * @param _how Array passed by Kernel when using 'authP()'. First item should be the address to check can perform.
+    * return True if the sender has not voted on the most recent open vote or closed unexecuted vote, false if they have.
+    */
+    function canPerform(address _who, address, bytes32, uint256[] _how) external view returns (bool) {
+        if (votesLength == 0) {
+            return true;
+        }
+
+        Vote storage latestVote = votes[votesLength];
+        address sender = _how.length > 0 ? address(_how[0]) : _who;
+
+        bool senderNotVotedOnLatestVote = lastYeaVoteId[sender] != votesLength;
+        bool latestVoteFinished = getBlockNumber64() > latestVote.startBlock.add(voteDurationBlocks);
+        bool latestVoteFailed = !_votePassed(latestVote);
+
+        return senderNotVotedOnLatestVote || (latestVoteFinished && (latestVoteFailed || latestVote.executed));
```

Comment:
@lkngtn Should the user be able to perform functions when the vote has been accepted, the execution delay period has elapsed but it has not been executed yet? As it stands it requires execution as well before this will return true. 

It currently returns true if the vote has finished and been rejected (which I believe satisfies the above requirement) OR the vote has finished, passed and been executed.

Reply:
Sorry for not clarifying.

> Should the user be able to perform functions when the vote has been accepted, the execution delay period has elapsed but it has not been executed yet? As it stands it requires execution as well before this will return true.

I think this part is Ok.

> It currently returns true if the vote has finished and been rejected

What i believe is missing from this part, is a check wether the delay period has passed or not.
If it has already passed, it shouldn't be able to perform functions.

Reply:
I think this requires clarification from @lkngtn. 

When you say delay period, I assume you are referring to the execution delay between the vote finishing and vote execution. In which case I don't see why someone who has voted yes on a failed vote would only have the window between the vote finishing and the end of the execution delay to perform an action (redeem() etc.), interested if you can see a reason. If this isn't the delay period you are referring too can you be more specific?